src/bpftrace.cpp

```diff
@@ -1237,10 +1261,18 @@ void BPFtrace::poll_output(bool drain)
       return;
     }
 
-    // Print all maps if we received a SIGUSR1 signal
     if (BPFtrace::sigusr1_recv) {
       BPFtrace::sigusr1_recv = false;
-      print_maps();
+
+      if (sigusr1_attached_probe_) {
+        if (run_special_probe(*sigusr1_attached_probe_,
+                              reinterpret_cast<trigger_fn_t>(signal_trigger))) {
+          LOG(ERROR) << "Failed to run signal probe";
+          return;
+        }
+      } else {
+        print_maps();
```

Comment:
I don't think that we should preserve the default behaviour here. It was just a convenience b/c some people wanted to inspect map contents during bpftrace script runtime but now that we have proper signal probes, they can just define it themselves.

I now realized that I confused you a lot b/c I missed this line in my earlier review and I though that we were dropping this behaviour. Which is why I insisted on the migration guide entry. My apologies about that.

Still, I think that we should drop it.

Reply:
Isn't it nice to keep it? For simple scripts it can be a nice bit of utility, you dont have to add the `signal{print(@)}` manually, which makes it a lttle bit simpler, and the code complexity of keeping this seems low.

Reply:
To me, it's a bit of a surprising behaviour, even though it is documented. Why should the tool print maps upon receiving some random signal? I understand that it was a convenient way to provide a requested feature but now that we have proper signal probes, I'd argue that we should stick to the principle of the least surprise.